src/backend/CommandBuffer.cpp

```diff
@@ -69,10 +69,23 @@ namespace backend {
             return true;
         }
 
-        bool ComputeTextureCopyBufferSize(CommandBufferBuilder*, const TextureCopyLocation& location, uint32_t* bufferSize) {
+        bool ComputeTextureCopyBufferSize(CommandBufferBuilder*, const TextureCopyLocation& location, uint32_t rowPitch, uint32_t* bufferSize) {
             // TODO(cwallez@chromium.org): check for overflows
-            uint32_t pixelSize = static_cast<uint32_t>(TextureFormatPixelSize(location.texture->GetFormat()));
-            *bufferSize = location.width * location.height * location.depth * pixelSize;
+            *bufferSize = rowPitch * location.height * location.depth;
```

Comment:
Depending on Ben's answer to the D3D12 placed footprint thread, this could be ((height - 1) * rowPitch + width) * depth. We would need to add a test for this corner case too.